src/main.cpp

```diff
@@ -15,12 +16,20 @@ namespace graft {
   void setHttpRouters(Manager& m);
 }
 
+static graft::GraftServer server;
+static void signal_handler_stop(int sig_num) {
+    LOG_PRINT_L0("Stoping server");
+    server.stop();
+}
+
 int main(int argc, const char** argv)
 {
+    std::signal(SIGINT, signal_handler_stop);
+    std::signal(SIGTERM, signal_handler_stop);
```

Comment:
I think it is better to set the handlers further, just before running the loop. Otherwise, we are stopping something that is not running yet, and we are not sure if it will run at all.

Nevertheless, I think I have better implementation that also fixes the server and manager interfaces, I'd like to discuss.